mgnify_pipelines_toolkit/analysis/shared/study_summary_generator.py

```diff
@@ -0,0 +1,263 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2024 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from collections import defaultdict
+import glob
+import logging
+from pathlib import Path
+
+import pandas as pd
+
+from mgnify_pipelines_toolkit.constants.db_labels import TAXDB_LABELS, ASV_TAXDB_LABELS
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+def parse_args():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-i",
+        "--indir",
+        required=True,
+        type=str,
+        help="Input directory to where all the individual analyses subdirectories are if summarising, or where all the summaries are if merging",
+    )
```

Comment:
I'd be tempted to add a util somewhere in the toolkit, for specifying argparse args must be a valid path:

```python
from pathlib import Path
def directory_path(path: str) -> Path:
    p = Path(path)
    if p.is_dir():
        return p
    raise NotADirectoryError(path)
```

then we can do 
```python
    parser.add_argument(
        "-i",
        "--indir",
        required=True,
        type=directory_path,
        help="Input directory to where all the individual analyses subdirectories are if summarising, or where all the summaries are if merging",
    )
```

Reply:
love this idea, thank you! will do it

Reply:
Well.. I think we should look into "click" https://click.palletsprojects.com/en/stable/
We can start composing command lines tools much faster (and the code is cleaner IMO)